src/parser/DASHTree.cpp

```diff
@@ -468,6 +468,40 @@ start(void *data, const char *el, const char **attr)
             }
             dash->currentNode_ |= DASHTree::MPDNODE_SEGMENTTEMPLATE;
           }
+		  else if (strcmp(el, "ContentProtection") == 0) //Inside Representation
```

Comment:
I usually create a function if I have nearly identical parse sequences.
beside this the issue is from theory not as simple as done here: Different representations can have different content-protection. This has to be at least checked here (and implemented in the future in main.cpp